web-client/public/js/update-app.js

```diff
@@ -14,7 +15,39 @@ const refreshApp = () => {
     }
 };
 
+const displayNetwork = (network, name) => {
+    uploadState.nodeColoring.reload(network, name);
+    if (document.getElementById("zoomSlider").disabled) {
+        document.getElementById("zoomSlider").disabled = false;
+    }
+
+    uploadState.currentNetwork = network;
+    console.log("Network: ", network); // Display the network in the console
+    $("#graph-metadata").html(network.genes.length + " nodes<br>" + network.links.length + " edges");
+
+    if (network.warnings.length > 0) {
+        displayWarnings(network.warnings);
+    }
+
+    $("#fileName").text(name); // Set the name of the file to display in the top bar
+    $("input[type='range']").off("input"); // I have no idea why I do this. Investigate later.
+
+    // If more things need to be turned off, we'll add them to this array
+    [ "#resetSliders", "#resetSlidersMenu", "#undoReset", "#undoResetMenu" ].forEach(
+        selector => $(selector).off("click")
+    );
+};
+
 export const updateApp = grnState => {
+    if (grnState.newNetwork) {
+        displayNetwork(grnState.network, grnState.name);
+        refreshApp();
+
+        // Rare exception to the MVC cycle: right now we have no way of knowing whether the network has changed
+        // (which is what necessitates displayNetwork), so we mark the model here.
+        grnState.newNetwork = false;
+    }
```

Comment:
Speaking of a "new network," here is some special-case logic that isn't fully MVC. Read the comment above and see if it makes sense.

I suspect, actually, that most of `displayNetwork` _is_ safe to call even when a network isn't freshly loaded or imported. So this special handling might become unnecessary eventually. But one step at a time.